Sources/StreamChat/Controllers/CurrentUserController/CurrentUserController_Tests.swift

```diff
@@ -102,16 +102,21 @@ final class CurrentUserController_Tests: StressTestCase {
         env.currentUserObserverItem = .mock(id: .unique)
 
         var synchronizeError: Error?
+        var synchronizeCalled = false
         controller.synchronize { [callbackQueueID] error in
             AssertTestQueue(withId: callbackQueueID)
             synchronizeError = error
+            synchronizeCalled = true
         }
 
         // Simulate connection successful
         client.simulateProvidedConnectionId(connectionId: .unique)
 
-        XCTAssertEqual(controller.state, .remoteDataFetched)
-        XCTAssertNil(synchronizeError)
+        AssertAsync {
+            Assert.willBeEqual(self.controller.state, .remoteDataFetched)
+            Assert.willBeTrue(synchronizeCalled)
+            Assert.willBeNil(synchronizeError)
```

Comment:
`synchronizeError` value is `nil` by default which means that this assert will succeed and even when the callback returns non-nil error, it won't be caught here. There's a couple of options:
- You can check the error is `nil` synchronously within the `synchronize` callback.
- You can change this to `staysNil`. It not 100% correct because it suggests the synchronize callback could be called multiple times.
- You can just do `XCTAssertNil(synchronizeError)` after the `AssertAsync { }` block.

Reply:
✅ Done